File: TinyScript/js/src/parser/ast/ReturnStmt.js
const Stmt = require('./Stmt')
const ASTNodeTypes = require('./ASTNodeTypes')
class ReturnStmt extends Stmt {
    constructor(){
        super(ASTNodeTypes.RETURN_STMT, "return")
    }
}

module.exports = ReturnStmt

const {Expr} = require('./index')
ReturnStmt.parse = (it) => {
    const lexeme = it.nextMatch("return")
    const expr = Expr.parse(it)
    const stmt = new ReturnStmt()
    stmt.setLexeme(lexeme)
    stmt.addChild(expr)
    return stmt;
}

File: TinyScript/java/src/main/java/parser/ast/ReturnStmt.java
package parser.ast;

import parser.util.ParseException;
import parser.util.PeekTokenIterator;

public class ReturnStmt extends Stmt {
    public ReturnStmt() {
        super(ASTNodeTypes.RETURN_STMT, "return");
    }

    public static ASTNode parse(PeekTokenIterator it) throws ParseException {
        var lexeme = it.nextMatch("return");
        var expr = Expr.parse(it);

        var stmt = new ReturnStmt();
        stmt.setLexeme(lexeme);
        if(expr != null) {
            stmt.addChild(expr);
        }
        return stmt;
    }


}
